domain/src/main/java/me/fontys/semester4/postalcode/PCConverter.java

```diff
@@ -0,0 +1,93 @@
+package me.fontys.semester4.postalcode;
+
+import com.healthmarketscience.jackcess.CursorBuilder;
+import com.healthmarketscience.jackcess.Database;
+import com.healthmarketscience.jackcess.DatabaseBuilder;
+import com.healthmarketscience.jackcess.IndexCursor;
+import me.fontys.semester4.data.entity.PostalcodePart;
+import me.fontys.semester4.data.repository.PostalcodePartRepository;
+import me.fontys.semester4.interfaces.FileConverter;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component("pcConverter")
+public class PCConverter implements FileConverter
+{
+    private final static String PC_TABLE="POSTCODES";
+    private final static String PC_RANGE_TABLE="GEMEENTEN";
+    private final static String PC_ROW_PC="A13_POSTCODE";
+    private final static String PC_ROW_RANGE_TYPE="A13_REEKSIND";
+    private final static String PC_ROW_RANGE_START="A13_BREEKPUNT_VAN";
+    private final static String PC_ROW_RANGE_END="A13_BREEKPUNT_TEM";
+    public final List<PostalcodePart> postalcodeParts;
+
+
+    public PCConverter()
+    {
+        postalcodeParts = new ArrayList<>();
+    }
+
+
+
+    @Override
+    public void Convert(File importFile) throws IOException
+    {
+        Database anImport = new DatabaseBuilder(importFile)
+                .setFileFormat(Database.FileFormat.V2000)
+                .open();
+
+        // check tables
+        for (var table: new String[] {PC_TABLE,PC_RANGE_TABLE})
+        {
+            if(!anImport.getTableNames().contains(table))
+            {
+                throw new UnsupportedEncodingException("Table missing: "+table);
+            }
+        }
+
+        // retrieve unique postal codes
+        IndexCursor cursor = CursorBuilder.createCursor(
+                anImport.getTable(PC_TABLE).getIndex("A13_POSTCODE"));
+
+        int counter=0,total=0;
+        for(var row: cursor)
+        {
+            // normalize postal code (no space)
+            var postalcode = row.getString(PC_ROW_PC).replace(" ","");
+
+            var even = row.getString(PC_ROW_RANGE_TYPE); // TODO: there is a type 2 and 3 as well..
+            var fromrange = row.getBigDecimal(PC_ROW_RANGE_START).longValueExact();
+            var torange = row.getBigDecimal(PC_ROW_RANGE_END).longValueExact();
+
+
+            //var ExistingPCs = postalcodePartRepository.findByPostalcodeAndEvenAndStarthousenrAndEndhousenr(
+            //        postalcode,even,fromrange,torange);
+
+            // Check if existing item in database and if not there, then add for importer.
+            //if(ExistingPCs.isEmpty())
```

Comment:
please remove commented code if it is not required anymore, VCS is the reason we can do this :)
p.s. also remove empty lines please :+1:

Reply:
The reason why it is not removed is because this is an initial version, and I still might need this when completing the final version.